traffic-advice.bs

```diff
@@ -0,0 +1,265 @@
+
+    1.  Let |parseResult| be the result of [=parsing traffic advice=] from |string| given |identity|.
+
+    1.  Run |whenComplete| with |parseResult|.
+
+1.  [=Fetch=] |request| with |processResponse| and |processResponseEndOfBody|.
+
+</section>
+
+Parsing {#parsing}
+==================
+
+<section algorithm="parse-traffic-advice">
+
+To <dfn>parse traffic advice</dfn> from a [=string=] |string| given [=agent identity=] |identity|:
+
+1.  Let |parsed| be the result of [=parsing JSON into Infra values=] given |string|. If this throws an exception, then return null.
+
+1.  If |parsed| is not a [=list=], then return null.
+
+1.  Let |bestMatch| be null.
+
+1.  [=list/For each=] |entry| of |parsed|:
+
+    1.  If |entry| is not a [=map=], then [=iteration/continue=].
+
+    1.  If |entry|[`"user_agent"`] does not [=map/exist=] or is not a [=string=], then [=iteration/continue=].
+
+    1.  Let |agentSelector| be |entry|[`"user_agent"`].
+
+    1.  If |identity| does not contain |agentSelector|, then [=iteration/continue=].
+
+    1.  If |bestMatch| is null or |agentSelector| appears at an earlier index in |identity| than |bestMatch|[`"user_agent"`], then set |bestMatch| to |entry|.
+
+1.  If |bestMatch| is null, then return null.
+
+1.  Let |entry| be a [=traffic advice entry=].
+
+1.  If |bestMatch|[`"disallowed"`] [=map/exists=] and is true, then set |entry|'s [=traffic advice entry/disallowed flag=] to true.
+
+1.  Return |entry|.
+
+</section>
+
+Interpretation {#interpretation}
+================================
+
+*This section is non-normative.*
+
+[=Traffic advice respecting agents=] should, when they would be able to respect advice to disallow traffic to an origin (for example, when requested to proxy prefetch traffic to the origin), [=fetch traffic advice=] (respecting HTTP caching semantics).
+
+If the result is null, then no advice was received and the agent should adopt its default behavior.
+
+If the result is `"unreachable"`, then the HTTP server was not able to service the request for traffic advice. The agent should disallow traffic for some interval if possible, since this may indicate that the server cannot accept additional requests at this time.
+
+If the result's [=traffic advice entry/disallowed flag=] is true, then the HTTP server advises that traffic should be disallowed at this time. Establishing new connections and sending requests should be avoided.
+
+Privacy and security considerations {#privacy-and-security}
+===========================================================
+
+<p class="issue">TODO: fill this in</p>
+
+IANA considerations {#iana}
+===========================
+
+Well-known `traffic-advice` URI {#iana-well-known}
+--------------------------------------------------
+
+This document defines well-known URI suffix `traffic-advice` as described by [[WELL-KNOWN]]. It should be submitted for registration as follows:
+
+:  URI suffix
+:: traffic-advice
+:  Change controller
+:: W3C
```

Comment:
This should probably be you personally (or maybe "The editor(s) of this document") until this gets an actual venue.

Reply:
Done.